### PR TITLE
Visual Improvements to API Docs

### PR DESCRIPTION
**Pull request by michaelzetune**
### Description

Visual improvements based on Sijia's [mock](https://github-production-user-asset-6210df.s3.amazonaws.com/30369272/255260757-f875d0f0-334d-429f-ab11-6834dd3535dd.png). See screenshots.

## Release notes

None.

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://docs.civiform.us/contributor-guide/developer-guide/internationalization-i18n#internationalization-for-application-strings)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

None.

### Instructions for manual testing

None.

### Issue(s) this completes

Relates to #5262.

### CODE REVIEW

File: server/app/views/api/ApiDocsView.java [hunk 1]
```diff
@@ -169,32 +172,47 @@ private DivTag programDocsDiv(ProgramDefinition programDefinition) {
   }
 
   private DivTag questionDocsDiv(QuestionDefinition questionDefinition) {
-    DivTag divTag = div().withClasses("pl-4", "border", "border-gray-500", "rounded-lg");
+    DivTag divTag =
+        div()
+            .withClasses("pl-4", "border-b", "border-gray-300", "pt-2", "pb-2", "flex", "flex-col");
 
-    divTag.with(
+    DivTag topHalf = div();
```
Comment (avaleske) [inline server/app/views/api/ApiDocsView.java:179]: here and elsewhere:
Thank you for updating these names to be more clear! Could we go a little further and say what this is the top half of, like `questionCardHeader`, `questionCardBodyLeft` and `questionCardBodyRight`?

**Review by avaleske — APPROVED** [review]
Out of scope for this, but how difficult would it be to add the top-level data (program id, submission time, etc) to the the sample response?

It would be good to clarify in the "How does this work?" section that the top-level data is missing in the sample response.

File: server/app/views/api/ApiDocsView.java (hunk 1 above)
Comment (michaelzetune) [reply server/app/views/api/ApiDocsView.java]: Done.

**Comment by michaelzetune** [comment]
> Out of scope for this, but how difficult would it be to add the top-level data (program id, submission time, etc) to the the sample response?
> 
> It would be good to clarify in the "How does this work?" section that the top-level data is missing in the sample response.

Very doable! It's the first unfinished TODO on #5262. Want to get to it by tomorrow.

**Pull request merged**